apps/ui/src/helpers/utils.ts

```diff
@@ -220,6 +220,15 @@ export function _d(s: number) {
     .trim();
 }
 
+export function toBigIntOrNumber(value) {
+  const parsedValue = parseFloat(value);
```

Comment:
This will introduce inaccuracies as `value` can be either number (for offchain) or string (for onchain), types were actually broken for a while (types say it should be number).

For example for some proposal we have scores_total "10000000000000000000000000".

<img width="363" alt="image" src="https://github.com/user-attachments/assets/5edf3619-0a43-40c0-bdca-9a44485d1c5b">

We can for time being instead check if `value` (let's add type annotation there as well):
1. is `string` -> run `BigInt(value)` - let it throw if it's invalid (contains decimal point) - this is onchain case
2. else (is `number` -> return as-is - this is offchain case.

We could make it better by fixing types for `quorum`/`scores_total`,`scores` in `Proposal`  entirely (they should be `number | string`. We will need to fix the types in few other places in the app, but in all those cases we can just convert them to `Number` and that will be backwards-compatible.

Reply:
Right now I am not doing: `BigInt(parseFloat(value))` 
It will just do `BigInt(value)` if value is integer, 

```ts
function toBigIntOrNumber(value) {
  const parsedValue = parseFloat(value);
  if (Number.isInteger(parsedValue)) {
    return BigInt(value);
  } else {
    return parsedValue;
  }
}
```

Agree with the confusion with types 🙈 maybe better to do it as a separate issue? 😅

Reply:
Looks like I misread the code, you are right.